src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java

```diff
@@ -283,12 +283,11 @@ private static DetailAST getWhitespaceFollowedNode(DetailAST ast) {
      * @return true if whitespace after ast should be checked
      */
     private static boolean shouldCheckWhitespaceAfter(DetailAST ast) {
-        boolean checkWhitespace = true;
         final DetailAST previousSibling = ast.getPreviousSibling();
-        if (previousSibling != null && previousSibling.getType() == TokenTypes.ANNOTATIONS) {
-            checkWhitespace = false;
-        }
-        return checkWhitespace;
+        final boolean isSynchronizedMethod = ast.getType() == TokenTypes.LITERAL_SYNCHRONIZED
```

Comment:
I think it is better to make it like
```
final boolean isSynchronizedBlock = ast.getType() == TokenTypes.LITERAL_SYNCHRONIZED 
 && ast.getFirstChild() != null;
return isSynchronizedBlock || previousSibling == null || previousSibling.getType() != TokenTypes.ANNOTATIONS;
```
looks clearer and parens are not needed

Reply:
Consider:

Current: !(A ^ B) ^ (C ^ !D)
Proposed: (A ^ !B) v C v !D

These are not equivalent; also short circuit of proposed expression causes some issues with array declarations.